.github/workflows/e2e-nvidia-l40s-x8.yml

```diff
@@ -58,6 +63,66 @@ jobs:
               {"Key": "GitHubPR", "Value": "${{ github.event.number }}"}
             ]
 
+      # Fallback in case of failure
+      - name: Attempt to start EC2 runner on us-east-2a (if us-east-2b failed)
+        if: steps.start-ec2-runner-us-east-2b.outcome == 'failure'
+        id: start-ec2-runner-us-east-2a
+        uses: machulav/ec2-github-runner@fcfb31a5760dad1314a64a0e172b78ec6fc8a17e # v2.3.6
+        continue-on-error: true
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ${{ vars.AWS_EC2_AMI }}
+          ec2-instance-type: ${{ env.INSTANCE_TYPE }}
+          subnet-id: subnet-02d230cffd9385bd4
+          security-group-id: sg-06300447c4a5fbef3
+          iam-role-name: instructlab-ci-runner
+          aws-resource-tags: >
+            [
+              {"Key": "Name", "Value": "instructlab-ci-github-xlarge-runner"},
+              {"Key": "GitHubRepository", "Value": "${{ github.repository }}"},
+              {"Key": "GitHubRef", "Value": "${{ github.ref }}"},
+              {"Key": "GitHubPR", "Value": "${{ github.event.number }}"}
+            ]
+
+      # Only try us-east-2c if the previous two availability zones had no availability, but
+      # do not continue if we see a failure here. Assume we've exhausted all options at that
+      # point in time.
+      - name: Attempt to start EC2 runner on us-east-2c (if us-east-2a failed)
+        if: steps.start-ec2-runner-us-east-2a.outcome == 'failure'
+        id: start-ec2-runner-us-east-2c
+        uses: machulav/ec2-github-runner@fcfb31a5760dad1314a64a0e172b78ec6fc8a17e # v2.3.6
+        with:
+          mode: start
+          github-token: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+          ec2-image-id: ${{ vars.AWS_EC2_AMI }}
+          ec2-instance-type: ${{ env.INSTANCE_TYPE }}
+          subnet-id: subnet-04701a08396b2ed01
+          security-group-id: sg-06300447c4a5fbef3
+          iam-role-name: instructlab-ci-runner
+          aws-resource-tags: >
+            [
+              {"Key": "Name", "Value": "instructlab-ci-github-xlarge-runner"},
+              {"Key": "GitHubRepository", "Value": "${{ github.repository }}"},
+              {"Key": "GitHubRef", "Value": "${{ github.ref }}"},
+              {"Key": "GitHubPR", "Value": "${{ github.event.number }}"}
+            ]
+
+      - name: Determine EC2 availability zone used
+        id: selected-availability-zone
+        run: |
+          # shellcheck disable=SC2086 # False positive error from shellcheck about word splitting/gobbling
+          if [ "${{ steps.start-ec2-runner-us-east-2b.outcome }}" = "success" ]; then
+            echo "label=${{ steps.start-ec2-runner-us-east-2b.outputs.label }}" >> $GITHUB_OUTPUT
+            echo "ec2-instance-id=\"${{ steps.start-ec2-runner-us-east-2b.outputs.ec2-instance-id }}" >> $GITHUB_OUTPUT
```

Comment:
This line is different from the others echoing ec2-instance-id - `ec2-instance-id=\"${{ ...` vs `ec2-instance-id=${{ ...`

They should all be the same, I think? But it's hard for me to guess just by looking whether we need the escaped quote at the beginning or not. However, if we do need the escaped quote at the beginning, we'd also need one at the end? Is this perhaps what the shellcheck warning was erroring about, and maybe it wasn't a false positive after all?

Reply:
ohhh, forgot to clean that up during testing. *doh* thanks! will remove the escaped chars